### PR TITLE
Weapons analyzer and modular guns rebalance

### PR DESCRIPTION
**Pull request by Sindorman**
This PR is pretty decent in size. The contents of changes are:
### Modular guns rebalance:

- Capacitors reliability was buffed: reinforced 80 -> 100, uranium 50 -> 60, teranium 45 -> 55, phoron 40 -> 50, bluespace 30 -> 45.

- Added nano reinforced barrel which has 10 more reliability than reinforced barrel.

- Added enhanced grip MK2, which has improved fire delay by 0.2 compared to MK1 grip.

- Added stablity stock, which improves fire delay by 0.2 and accuracy by 0.5 compared to regular stock.

- Added animations to lots of modular components like all capacitors, almost all modulators, lenses, etc.

- Modular guns now have to be assembled inside of weapons analyzers. You cannot assemble them in hands anymore

### Weapons analyzer:

- Added machinery called Weapons analyzer. Main purpose of it is to display stats of any weapon or item. You can accepts anything that is an item, even trash(probably). For modular guns it will also display table of components and their statistics.

- Modular guns now need to be assembled inside of weapons analyzer.

- Added circuit board and ability to construct weapons analyzer in RND. Circuit board required tech: engineering = 3, data = 2, combat = 2.

### Misc changes:

- recharger and wall recharger now visually display on their icon how much charge is inside of weapon/gun that is being charged.

Screenshots:

Item:
ErbvLurdmh

Gun:
pJNhiVeQpD

Modular gun:
JBYIW1IUak

8iDVEqqNDB

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

**Comment by TehFlaminTaco** [comment]
Force is a magic number and technically meta information. I recommend changing it up a little to display less raw numbers and more "General areas"

File: vueui/dist/app.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
...laqtdQgUKMdQ7GS/ER8rd+san1XAClAY1f0lhvVxoKBtChaj9/WXBTw7hNhVeDzNAJquIyYPUqBl+PdiDQp1LdRe2tekS7Z0u669YbF2NrxBINuWWr+x/wAilrpn14gHShqI6AaCXHIGQ19YdmLaOWXxt8JdsWt8HpLmhc2Y8njth8ovLiaO1cwqWizLzszFOvq8QSuGc+8bUpz38S5y8uJsVBHS8A4OCPgfOD9xpYVp++ox2rivaVSp8+o2QXj0+ZQgPkvXcxVc5Zcg1q/1DCic/aUxR5ESgKB1q4SqwZDn4gx78ECWWcF6ggQQFg3Sz9wsTy91KNGVhZWA4M5zM1zKy8RVwAZZqhpn1mdw/wBRRUKeI45wFB/eYABW8TMvQc9wkiqx9ZZyivPsQUxhl/m4FBxefSDSr75ZSJvt5+IxLYWn14lEGqqEBbWaPdl1HFYOfWWldmCWtbdB7R/c8fzmZpmwPSFqWrlzCNNu/wBSuzLwt/5EwvrDAFBlgik5f1yoVlk7mQgrxGZeqoghRbs4QBGnwzpjN2qfvAFm3n2hk3jKeIpFhzfXcCrADd89/wCy4qh28v6j3JBj1eJscAiRD4DqUC2vB7xHC3xEVWsQLNzXpLyK/wDEpW1bu9dx5ucrUPAp5gi1n59pcWXkX2lbtjKwpRRzdX6ygE8+4zG31b6mSWg4+IibaDRxNIx4P3Li5WuUtIUYolARmfZ4jYBrvj6cxsNqbRllZ5zm3mUOg2xqho6OIhAUHXPrANTTR5/5MhLujxfpEBNt0ddS0MdyGbo/4hXoV/KVve11+DqEKF5y/Uzk421CsUAW/uOclX7+selx/tAKrQf2JSKtVjqZFmv5gRKsYJY5OD6wbsnuXFx8n+blTLcv12wsUI2rm5Ru2y8zIBjnnqXJVRNlgc/7GTw+IKc+DWYqha4OuoABbHGvmCv+H1hUDL5c9woYvyvmUWinC8+kZGhtr+zFaqDaXqsHTzGebyp5myujBX+S0+GA4lW0tLDghAGyaq1OYYbLfpxNwzp669YHiM7eJScMZX/Y1l3y8/ENtA9b6iBeBKr9R9gOiEhFbB+4Au7u6OYqP2fqLnSx4NTLALPsQsFl3+5Rg54f2ISKxdLz0TLAgczS8desCBmv8gIpy3i+4ycA/LGrd0z6sYlQBX/CYkcNH95mQDZYfuBZcjHJlkstiADM
```
Comment (TehFlaminTaco) [inline vueui/dist/app.css:4]: Do not include the dist folder in your PR (I recommend making another Commit with the old dist folder).
Vue is automatically built via our CI.

File: code/modules/research/weaponsanalyzer.dm [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+	return TRUE
+
+/obj/machinery/weapons_analyzer/verb/eject()
+	set name = "Eject Inserted Item"
+	set category = "Object"
+	set src in view(1)
+
+	if(!usr || usr.stat || usr.lying || usr.restrained() || !Adjacent(usr))
+		return
```
Comment (NonQueueingMatt) [inline code/modules/research/weaponsanalyzer.dm:110]: you wanna use use_check_and_message here

File: code/modules/research/weaponsanalyzer.dm [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+	
+	if(gun)
+		gun.forceMove(usr.loc)
```
Comment (NonQueueingMatt) [inline code/modules/research/weaponsanalyzer.dm:113]: should probably use get_turf(usr) here, or try putting it in their hands

Comment (TehFlaminTaco) [reply code/modules/research/weaponsanalyzer.dm:113]: or `get_turf(this)` if you want to behave like some other machines.

File: code/modules/research/weaponsanalyzer.dm [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+	icon_state = "weapon_analyzer"
+	density = 1
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 60
+	active_power_usage = 2000
+
+	var/obj/item/gun/gun = null
+	var/obj/item/item = null
+	var/obj/item/device/laser_assembly/assembly = null
```
Comment (NonQueueingMatt) [inline code/modules/research/weaponsanalyzer.dm:13]: Would it not be better to just have one variable cast as obj then typecast further when required?

File: code/modules/research/weaponsanalyzer.dm (hunk 3 above)
Comment (Sindorman) [reply code/modules/research/weaponsanalyzer.dm:113]: That would drop it under the machine, and then you won't see it

**Comment by Sindorman** [comment]
> Force is a magic number and technically meta information. I recommend changing it up a little to display less raw numbers and more "General areas"

Force is an abstract number indeed. But that is the only number that will help you approximate damage from an item. It is not possible to calculate exact damage it will deal, especially since someone can tweak formula for calculations later and not update analyzer. I do not think it is a meta information, as it literally means how much maximum force can a person exert with this object

File: code/modules/research/weaponsanalyzer.dm (hunk 3 above)
Comment (TehFlaminTaco) [reply code/modules/research/weaponsanalyzer.dm]: Pretty sure `/obj/machinery` by default is on a layer below `/obj/item`

**Comment by NonQueueingMatt** [comment]
i'm not a fan of how data is presented, it needs to be less magic numbers directly given and more an actual flavor string

look at health analyzers in medical for an example

**Comment by Sindorman** [comment]
Updated UI's now give information in less meta way.
C5SiHDBaTo
s1UXflVLNQ

**Pull request merged**